### PR TITLE
Graph: Mimic quirk for sln-based builds with target name containing semicolon

### LINKED ISSUE #9376: [Bug]: Solution-based graph builds don't handle a quoted target list consistent with non-graph builds
### Issue Description

When using a quoted target list (eh `/t:"Clean;Build"`), the behavior is inconsistent with non-graph builds.

> error MSB4057: The target "Clean;Build" does not exist in the project.

MSBuild itself also behaves a bit inconsistently, the that's potentially a separate bug (@rainersigwald?).

Setup:
```
mkdir QuotedTargetRepro
cd QuotedTargetRepro
dotnet new classlib
dotnet new sln
dotnet sln add .
msbuild /t:restore
```

Non-graph
* `msbuild /v:m /graph:false /t:Clean;Build QuotedTargetRepro.sln` - WORKS
* `msbuild /v:m /graph:false /t:"Clean;Build" QuotedTargetRepro.sln` - WORKS
* `msbuild /v:m /graph:false /t:Clean;Build QuotedTargetRepro.csproj` - WORKS
* `msbuild /v:m /graph:false /t:"Clean;Build" QuotedTargetRepro.csproj` - FAIL

Graph:

* `msbuild /v:m /graph /t:Clean;Build QuotedTargetRepro.sln` - WORKS
* `msbuild /v:m /graph /t:"Clean;Build" QuotedTargetRepro.sln` - FAIL <--- This bug
* `msbuild /v:m /graph /t:Clean;Build QuotedTargetRepro.csproj` - WORKS
* `msbuild /v:m /graph /t:"Clean;Build" QuotedTargetRepro.csproj` - FAIL

### Steps to Reproduce

See above

### Expected Behavior

See above

### Actual Behavior

See above

### Analysis

_No response_

### Versions & Configurations

_No response_

### PR DESCRIPTION
**Pull request by dfederm**
Fixes #9376

There is a quirk for solution builds for targets containing semicolons. For example if you do a solution build using `/t:"Clean;Build"`, the metaproject ends up calling the `<MSBuild>` task using that value, which ends up splitting it into two targes: "Clean" and "Build". However if you try the same thing on a project, it will actually end up attempting to call a target named "Clean;Build", which would also certainly fail in practice (although in theory there could be a distinct target with that crazy name).

This change just mimics that quirk for graph builds.

See linked issue for a practical repro and test cases.

### CODE REVIEW

File: src/Build/Graph/GraphBuilder.cs [hunk 1]
```diff
@@ -257,8 +259,7 @@ private static void AddEdgesFromSolution(IReadOnlyDictionary<ConfigurationMetada
                         string.Join(";", entryPoints.Select(e => e.ProjectFile))));
             }
 
-            ErrorUtilities.VerifyThrowArgument(entryPoints.Count == 1, "StaticGraphAcceptsSingleSolutionEntryPoint");
```
Comment (dfederm) [inline src/Build/Graph/GraphBuilder.cs:260]: Note: removed this as the exact same check is done above already. Also, this error message requires a param, which this is not providing, so even if this code were hit it'd result in incorrect behavior.

File: src/Build/Graph/ProjectGraph.cs [hunk 2]
```diff
@@ -56,9 +56,11 @@ public sealed class ProjectGraph
 
         private readonly Lazy<IReadOnlyCollection<ProjectGraphNode>> _projectNodesTopologicallySorted;
 
-        private GraphBuilder.GraphEdges Edges { get; }
+        private readonly bool _isSolution;
 
-        internal GraphBuilder.GraphEdges TestOnly_Edges => Edges;
+        private readonly GraphBuilder.GraphEdges _edges;
```
Comment (dfederm) [inline src/Build/Graph/ProjectGraph.cs:61]: Unrelated minor change: Changed `Edges` from a property to a field since it was private.

**Review by maridematte — APPROVED** [review]

**Review by Forgind — APPROVED** [review]

File: src/Build/Graph/ProjectGraph.cs [hunk 3]
```diff
@@ -598,6 +601,24 @@ private string DebuggerDisplayString()
...
+                foreach (string entryTarget in entryProjectTargets)
+                {
+                    foreach (string s in ExpressionShredder.SplitSemiColonSeparatedList(entryTarget))
+                    {
+                        newEntryTargets.Add(s);
+                    }
+                }
+
+                entryProjectTargets = newEntryTargets;
+            }
```
Comment (JanKrivanek) [inline src/Build/Graph/ProjectGraph.cs:620]: Why recreating the list twice? (`SplitSemiColonSeparatedList` already creates new)

```suggestion
            {
                entryProjectTargets = ExpressionShredder.SplitSemiColonSeparatedList(entryTarget);
            }
```

**Review by JanKrivanek — APPROVED** [review]
Looks good.

I have a minor comment for shortening the code

**Comment by yuehuang010** [comment]
FYI, long long time ago, I added a funky feature to address solution building multiple targets, non-graph.  Looks up SolutionProjectReferenceAllTargets or SlnProjectResolveProjectReference.  I wasn't able to enable it because there was a high chance of a regression due to missing P2P references with a different build order.  With graph build, this turns over a new leaf so perhaps consider addressing the issue directly.

**Comment by JanKrivanek** [comment]
> FYI, long long time ago, I added a funky feature to address solution building multiple targets, non-graph. Looks up SolutionProjectReferenceAllTargets or SlnProjectResolveProjectReference. I wasn't able to enable it because there was a high chance of a regression due to missing P2P references with a different build order. With graph build, this turns over a new leaf so perhaps consider addressing the issue directly.

@yuehuang010 - this one?: https://github.com/dotnet/msbuild/pull/7512

**Comment by AR-May** [comment]
Team triage: It seems to be resolved via PR #9452. We decided to close this one in favor to #9452.
